backend/curation/api/v1/curation/collections/common.py

```diff
@@ -33,15 +33,14 @@ def get_collections_base_url():
 
 
 def extract_dataset_assets(dataset_version: DatasetVersion):
-    base_url = CorporaConfig().dataset_assets_base_url
     asset_list = list()
     for asset in dataset_version.artifacts:
         if asset.type not in allowed_dataset_asset_types:
             continue
         filesize = get_business_logic().s3_provider.get_file_size(asset.uri)
         if filesize is None:
             filesize = -1
-        url = f"{base_url}/{dataset_version.version_id.id}.{asset.type}"
+        url = get_business_logic().generate_permanent_url(dataset_version.version_id, asset.type)
```

Comment:
If `generate_permanent_url` is a static method, you may want to do `BusinessLogic.generate_permanent_url`.  Alternatively, it could be a method of `CollectionVersionBase` (more OOP compliant), or be in `helpers.py`.

Reply:
This is also fine btw.